Userland/Libraries/LibCore/EventLoop.h

```diff
@@ -141,6 +146,7 @@ class EventLoop {
     };
 
     Vector<QueuedEvent, 64> m_queued_events;
+    NonnullRefPtrVector<Promise<NonnullRefPtr<Object>>> m_pending_promises;
```

Comment:
It seems the placement of this member caused some alignment issues with the private member? That really doesn't make sense, unless some other code in here or in the referenced test does funny things with EventLoops?
https://github.com/SerenityOS/serenity/actions/runs/4297556898/jobs/7490672728#step:25:423

>   Running test 'local_socket_write'.
  UBSAN: member access within misaligned address 0x0000000063fe87db of type 'struct Private' which requires 4 byte alignment
  UBSAN: at /home/runner/work/serenity/serenity/Userland/Libraries/LibCore/EventLoop.cpp, line 524, column: 44
  UB is configured to be deadly
  ASSERTION FAILED: false
  ./Userland/Libraries/LibSanitizer/UBSanitizer.cpp:46

Reply:
I'm not sure since I ran these exact tests locally on both toolchains without problems.

Reply:
Did you run them with UBSAN enabled?

Reply:
I think I have a reproduction but I need to investigate a little more. It's pretty flaky.